core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -309,6 +328,48 @@ private void releaseBlockLockIfUnused(long blockId) {
   }
 
   /**
+   * Checks the internal state of the manager to make sure invariants hold.
+   *
+   * This method is intended for testing purposes. A runtime exception will be thrown if invalid
+   * state is encountered.
+   */
+  public void validate() {
```

Comment:
I'm more worried about the maintenance cost of using reflection than I am that someone will call this method in production code (it might even make sense to do this for some sort of "validateSystem" endpoint). This method also serves as a sort of documentation about the invariants we expect in this file. I agree with using reflection to avoid exposing private state, but I think we shouldn't push the power of reflection too far.